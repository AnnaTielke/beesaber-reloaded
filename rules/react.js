module.exports = {
  plugins: ['react'],

  extends: ['plugin:react/recommended'],

  settings: {
    react: {
      version: 'detect',
    },
  },

  rules: {
    'react/button-has-type': 'error',
    'react/jsx-boolean-value': 'error',
    'react/jsx-fragments': 'error',
    'react/jsx-handler-names': 'error',
    'react/jsx-no-bind': ['error', { allowArrowFunctions: true }],
    'react/jsx-pascal-case': 'error',
    'react/no-access-state-in-setstate': 'error',
    'react/no-array-index-key': 'error',
    'react/no-danger': 'error',
    'react/no-did-mount-set-state': 'error',
    'react/no-did-update-set-state': 'error',
    'react/no-multi-comp': 'error',
    'react/no-this-in-sfc': 'error',
    'react/no-typos': 'error',
    'react/no-unsafe': 'error',
    'react/no-unused-prop-types': 'error',
    'react/no-unused-state': 'error',
    'react/no-will-update-set-state': 'error',
    'react/prefer-es6-class': 'error',
    'react/prefer-stateless-function': 'error',
    'react/style-prop-object': 'error',
    'react/void-dom-elements-no-children': 'error',
  },
};
