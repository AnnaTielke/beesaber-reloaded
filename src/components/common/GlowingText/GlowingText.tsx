import React from 'react';
import styled from 'styled-components';

import colors, { allColorSetsObject } from '../../../tokens/definitions/color';

import type { ColorSet, ColorStrings } from '../../../tokens/definitions/color';

const Element = styled('span')<{ titleColor: ColorSet }>`
  ${({ titleColor }) => colors.createGlow(titleColor)}
`;

type ElementType = 'h1' | 'h2' | 'h2' | 'p' | 'span';

type Props = {
  as?: ElementType;
  children: React.ReactNode;
  titleColor: ColorStrings;
};

const GlowingText = ({
  as = 'span',
  children,
  titleColor,
}: Props): JSX.Element => (
  <Element as={as} titleColor={allColorSetsObject[titleColor]}>
    {children}
  </Element>
);

export type { ElementType };

export default GlowingText;
