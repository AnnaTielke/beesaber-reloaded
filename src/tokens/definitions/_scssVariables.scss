// F O N T S //

@font-face {
  font-family: 'NeonTubes2';
  src:
    url('../definitions/neonTubes/neontubes2-webfont.woff') format('woff'),
    url('../definitions/neonTubes/NeonTubes2.otf') format('opentype'),
    url('../definitions/neonTubes/NeonTubes2.otf') format('truetype');
}

// C O L O R S
$bg-color: #00111f;    // Page Background Color
$main-red: #ff0025;    // Beat Saber Red
$main-red-light: #fff0f0;    // Beat Saber Red Light
$main-red-shadow: #ff8d9e;    // Beat Saber Red Shadow
$main-blue: rgb(0 0 255);       // Beat Saber Blue
$main-blue-light: #edf5ff;    // Beat Saber Blue Light
$main-blue-shadow: #7979ff;    // Beat Saber Blue Shadow
$main-yellow: #ffeb00;    // Bee Saber Yellow
$main-yellow-light: #fffcdd;    // Bee Saber Yellow Light (for Shadow)
$main-white: rgb(245 245 245); // Bee Saber White
$main-white-light: #747373;    // Bee Saber White-light
$main-grey: rgb(128 128 128);       // Bee Saber Grey
$main-grey-light: #ced4da;    // Bee Saber Grey-light
$main-grey-dark: #131c24;    // Bee Saber Grey-dark
