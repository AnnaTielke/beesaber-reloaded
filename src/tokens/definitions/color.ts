type ColorSet = {
  light: string;
  main: string;
  shadow?: string;
};

type PageSet = {
  bgColor: string;
};

const page: PageSet = {
  bgColor: '#00111f',
};

const blue: ColorSet = {
  main: 'blue',
  light: '#edf5ff',
  shadow: '#7979ff',
};

const red: ColorSet = {
  main: '#ff0025',
  light: '#fff0f0',
  shadow: '#ff8d9e',
};

const yellow: ColorSet = {
  main: '#ffeb00',
  light: '#fffcdd',
};

const createGlow = (color: ColorSet): string => `
  color: ${color.light};
  -webkit-transition: color 0.6s ease-out, text-shadow 0.6s ease-out;
  -moz-transition: color 0.6s ease-out, text-shadow 0.6s ease-out;
  -o-transition: color 0.6s ease-out, text-shadow 0.6s ease-out;
  transition: color 0.6s ease-out, text-shadow 0.6s ease-out;
  text-shadow: ${color.main} -2px -1px 5px, ${color.main} 2px 1px 5px,
    ${color.main} 0px 0px 10px, ${color.main} 0px 0px 15px, ${color.main} 0px 0px 20px,
    ${color.main} 0px 0px 30px;
`;
const allColorSetsArray = [blue, red, yellow];

export { allColorSetsArray };

export type { ColorSet, PageSet };

export default {
  blue,
  createGlow,
  page,
  red,
  yellow,
};
