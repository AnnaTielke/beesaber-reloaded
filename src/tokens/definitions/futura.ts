const definitions = {
  // https://fonts.adobe.com/fonts/futura-pt#details-section
  book: {
    fontFamily: 'futura-pt, sans-serif',
    fontWeight: 400,
    fontStyle: 'normal',
  },
  demi: {
    fontFamily: 'futura-pt, sans-serif',
    fontWeight: 600,
    fontStyle: 'normal',
  },
} as const;

export default definitions;
