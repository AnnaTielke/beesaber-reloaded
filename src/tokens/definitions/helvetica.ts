export default "'Helvetica Neue', Helvetica, Arial, 'Open Sans', sans-serif";
