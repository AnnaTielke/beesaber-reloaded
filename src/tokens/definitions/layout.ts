type MediaSize = 'lg' | 'md' | 'mobile' | 'sm' | 'xl';

type BreakPointSize = '0px' | '575px' | '576px' | '768px' | '1024px' | '1680px';

type Breakpoints = Record<MediaSize, BreakPointSize>;

const breakpoints: Breakpoints = {
  mobile: '575px',
  sm: '576px',
  md: '768px',
  lg: '1024px',
  xl: '1680px',
};

type MediaQuery = Record<MediaSize, string>;

const mediaQuery: MediaQuery = {
  mobile: `@media (max-width: ${breakpoints.mobile})`,
  sm: `@media (min-width: ${breakpoints.sm})`,
  md: `@media (min-width: ${breakpoints.md})`,
  lg: `@media (min-width: ${breakpoints.lg})`,
  xl: `@media (min-width: ${breakpoints.xl})`,
};

const layout = {
  breakpoints,
  mediaQuery,
};

export { breakpoints, mediaQuery };

export default layout;
