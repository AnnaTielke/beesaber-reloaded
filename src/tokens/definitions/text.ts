import futura from './futura';
import helvetica from './helvetica';

const futuraBook = futura.book;
const futuraDemi = futura.demi;

const definitions = {
  display: {
    ...futuraDemi,
    fontSize: '4rem',
    lineHeight: '1.05',
  },
  headline1: {
    ...futuraDemi,
    fontSize: '2.5rem',
    lineHeight: '1.2',
  },
  headline2: {
    ...futuraDemi,
    fontSize: '1.75rem',
    lineHeight: '1.15',
  },
  headline3: {
    ...futuraDemi,
    fontSize: '1.375rem',
    lineHeight: '1.1',
  },
  headline4: {
    ...futuraDemi,
    fontSize: '1.125rem',
    lineHeight: '1.35',
  },
  info: {
    regular: {
      ...futuraBook,
      fontSize: '0.8125rem',
      lineHeight: '1.25',
    },
    strong: {
      ...futuraDemi,
      fontSize: '0.8125rem',
      lineHeight: '1.25',
    },
  },
  base: {
    regular: {
      fontFamily: helvetica,
      fontSize: '0.875rem',
      fontWeight: 400,
      lineHeight: '1.25',
    },
    strong: {
      fontFamily: helvetica,
      fontSize: '0.875rem',
      fontWeight: 500,
      lineHeight: '1.25',
    },
  },
  body: {
    regular: {
      fontFamily: helvetica,
      fontSize: '1rem',
      fontWeight: 400,
      lineHeight: '1.5',
    },
    strong: {
      fontFamily: helvetica,
      fontSize: '1rem',
      fontWeight: 500,
      lineHeight: '1.5',
    },
  },
} as const;

export default definitions;
