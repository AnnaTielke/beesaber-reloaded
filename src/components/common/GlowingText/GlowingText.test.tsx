import { render } from '@testing-library/react';
import React from 'react';

import { colorStringsArray } from '../../../tokens/definitions/color';
import GlowingText from './GlowingText';

import type { ElementType } from './GlowingText';

describe('GlowingText', () => {
  it.each(colorStringsArray)(
    'should match the snapshot when titlecolor is %s',
    (titleColor) => {
      const { container } = render(
        <GlowingText titleColor={titleColor}>Glowing Text</GlowingText>
      );

      expect(container.firstChild).toMatchSnapshot();
    }
  );

  it.each(['h1', 'h2', 'h2', 'p', 'span', undefined] as const)(
    'should match the snapshot when element is %s',
    (as: ElementType | undefined) => {
      const { container } = render(
        <GlowingText as={as} titleColor={'red'}>
          Glowing Text
        </GlowingText>
      );

      expect(container.firstChild).toMatchSnapshot();
    }
  );
});
