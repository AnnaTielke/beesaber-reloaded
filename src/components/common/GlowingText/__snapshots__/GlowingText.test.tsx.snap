// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlowingText should match the snapshot when element is h1 1`] = `
.c0 {
  color: #fff0f0;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -moz-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -o-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  text-shadow: #ff0025 -2px -1px 5px,#ff0025 2px 1px 5px, #ff0025 0px 0px 10px,#ff0025 0px 0px 15px,#ff0025 0px 0px 20px, #ff0025 0px 0px 30px;
  font-family: 'NeonTubes2';
}

<h1
  class="c0"
>
  Glowing Text
</h1>
`;

exports[`GlowingText should match the snapshot when element is h2 1`] = `
.c0 {
  color: #fff0f0;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -moz-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -o-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  text-shadow: #ff0025 -2px -1px 5px,#ff0025 2px 1px 5px, #ff0025 0px 0px 10px,#ff0025 0px 0px 15px,#ff0025 0px 0px 20px, #ff0025 0px 0px 30px;
  font-family: 'NeonTubes2';
}

<h2
  class="c0"
>
  Glowing Text
</h2>
`;

exports[`GlowingText should match the snapshot when element is h2 2`] = `
.c0 {
  color: #fff0f0;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -moz-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -o-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  text-shadow: #ff0025 -2px -1px 5px,#ff0025 2px 1px 5px, #ff0025 0px 0px 10px,#ff0025 0px 0px 15px,#ff0025 0px 0px 20px, #ff0025 0px 0px 30px;
  font-family: 'NeonTubes2';
}

<h2
  class="c0"
>
  Glowing Text
</h2>
`;

exports[`GlowingText should match the snapshot when element is p 1`] = `
.c0 {
  color: #fff0f0;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -moz-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -o-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  text-shadow: #ff0025 -2px -1px 5px,#ff0025 2px 1px 5px, #ff0025 0px 0px 10px,#ff0025 0px 0px 15px,#ff0025 0px 0px 20px, #ff0025 0px 0px 30px;
  font-family: 'NeonTubes2';
}

<p
  class="c0"
>
  Glowing Text
</p>
`;

exports[`GlowingText should match the snapshot when element is span 1`] = `
.c0 {
  color: #fff0f0;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -moz-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -o-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  text-shadow: #ff0025 -2px -1px 5px,#ff0025 2px 1px 5px, #ff0025 0px 0px 10px,#ff0025 0px 0px 15px,#ff0025 0px 0px 20px, #ff0025 0px 0px 30px;
  font-family: 'NeonTubes2';
}

<span
  class="c0"
>
  Glowing Text
</span>
`;

exports[`GlowingText should match the snapshot when element is undefined 1`] = `
.c0 {
  color: #fff0f0;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -moz-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -o-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  text-shadow: #ff0025 -2px -1px 5px,#ff0025 2px 1px 5px, #ff0025 0px 0px 10px,#ff0025 0px 0px 15px,#ff0025 0px 0px 20px, #ff0025 0px 0px 30px;
  font-family: 'NeonTubes2';
}

<span
  class="c0"
>
  Glowing Text
</span>
`;

exports[`GlowingText should match the snapshot when titlecolor is blue 1`] = `
.c0 {
  color: #edf5ff;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -moz-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -o-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  text-shadow: blue -2px -1px 5px,blue 2px 1px 5px, blue 0px 0px 10px,blue 0px 0px 15px,blue 0px 0px 20px, blue 0px 0px 30px;
  font-family: 'NeonTubes2';
}

<span
  class="c0"
>
  Glowing Text
</span>
`;

exports[`GlowingText should match the snapshot when titlecolor is red 1`] = `
.c0 {
  color: #fff0f0;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -moz-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -o-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  text-shadow: #ff0025 -2px -1px 5px,#ff0025 2px 1px 5px, #ff0025 0px 0px 10px,#ff0025 0px 0px 15px,#ff0025 0px 0px 20px, #ff0025 0px 0px 30px;
  font-family: 'NeonTubes2';
}

<span
  class="c0"
>
  Glowing Text
</span>
`;

exports[`GlowingText should match the snapshot when titlecolor is yellow 1`] = `
.c0 {
  color: #fffcdd;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -moz-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -o-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  text-shadow: #ffeb00 -2px -1px 5px,#ffeb00 2px 1px 5px, #ffeb00 0px 0px 10px,#ffeb00 0px 0px 15px,#ffeb00 0px 0px 20px, #ffeb00 0px 0px 30px;
  font-family: 'NeonTubes2';
}

<span
  class="c0"
>
  Glowing Text
</span>
`;
